upstream/controllers.go

```diff
@@ -28,17 +31,31 @@ func PushUpstream(c echo.Context) error {
 		return c.JSON(http.StatusBadRequest, api.HTTPError{Error: "cluster_name name is required", Message: "cluster name is required"})
 	}
 
-	headlessTplID, ok := templateIDCache.Get(req.ClusterName)
+	agentID, ok := agentIDCache.Get(req.ClusterName)
```

Comment:
This would require a unique constraint on name

Also, lets change the name from ClusterName to AgentName ? Since it can work from non-cluster environments too